sqlmesh/core/plan/definition.py

```diff
@@ -177,7 +179,7 @@ def modified_snapshots(self) -> t.Dict[SnapshotId, Snapshot]:
     @property
     def start(self) -> TimeLike:
         """Returns the start of the plan or the earliest date of all snapshots."""
-        if not self.override_start and (self._restate_models or not self._missing_intervals):
+        if not self._start and (self._restate_models or not self._missing_intervals):
```

Comment:
@eakmanrq we should not rely on `override_start` because sometimes `_start` is set to `default_start` in which case the `override_start` flag won't be set. At the same time we should still use the existing value of `_start`.